tests/transforms/test_qcut.py

```diff
@@ -1959,6 +1962,79 @@ def test_mc_state_prep(self):
                 assert op.wires == exp_op.wires
 
 
+class TestMCPostprocessing:
+    """
+    Tests that the postprocessing for circuits containing sample measurements
+    gives the correct results.
+    """
+
+    def test_sample_postprocess(self):
+        """
+        Tests that the postprocessing for the generic sampling case gives the
+        correct result
+        """
+
+        fragment_tapes = [frag0, frag1]
+
+        communication_graph = MultiDiGraph(frag_edge_data)
+        shots = 3
+        fragment_configurations, settings = qml.transforms.qcut.expand_fragment_tapes_mc(
+            fragment_tapes, communication_graph, shots
+        )
+
+        fixed_samples = [
+            np.array([[1.0], [0.0], [1.0], [1.0]]),
+            np.array([[0.0], [0.0], [1.0], [-1.0]]),
+            np.array([[0.0], [1.0], [1.0], [-1.0]]),
+            np.array([[0.0], [-1.0], [1.0]]),
+            np.array([[0.0], [-1.0], [-1.0]]),
+            np.array([[1.0], [1.0], [1.0]]),
+        ]
+
+        postprocessed = qcut.qcut_processing_fn_sample(fixed_samples, communication_graph, shots)
+
+        expected_postprocessed = [np.array([[1.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 1.0, 1.0]])]
+
+        for pp, exp_pp in zip(postprocessed, expected_postprocessed):
+            assert np.allclose(pp, exp_pp)
+
+    def test_mc_sample_postprocess(self):
+        """
+        Tests that the postprocessing for the generic sampling case gives the
+        correct result
+        """
+
+        fragment_tapes = [frag0, frag1]
+
+        communication_graph = MultiDiGraph(frag_edge_data)
+        shots = 3
+        fragment_configurations, settings = qml.transforms.qcut.expand_fragment_tapes_mc(
+            fragment_tapes, communication_graph, shots
+        )
+
+        fixed_samples = [
+            np.array([[1.0], [0.0], [1.0], [1.0]]),
+            np.array([[0.0], [0.0], [1.0], [-1.0]]),
+            np.array([[0.0], [1.0], [1.0], [-1.0]]),
+            np.array([[0.0], [-1.0], [1.0]]),
+            np.array([[0.0], [-1.0], [-1.0]]),
+            np.array([[1.0], [1.0], [1.0]]),
+        ]
+
+        def func(bitstring):
+            return np.sum(bitstring)
+
+        fixed_settings = np.array([[0, 7, 1], [5, 7, 2], [1, 0, 3], [5, 1, 1]])
+
+        postprocessed = qcut.qcut_processing_fn_mc(
+            fixed_samples, communication_graph, fixed_settings, shots, func
+        )
+
+        expected = 85.33333333333333
+
+        assert np.isclose(postprocessed, expected)
+
+
```

Comment:
Perhaps worth adding a quick test for `_reshape_and_find_degrees`.

Reply:
Added thanks!